src/config.c

```diff
@@ -473,8 +473,17 @@ void loadServerConfigFromString(char *config) {
         } else if ((!strcasecmp(argv[0],"slaveof") ||
                     !strcasecmp(argv[0],"replicaof")) && argc == 3) {
             slaveof_linenum = linenum;
+            if (!strcasecmp(argv[1], "no") && !strcasecmp(argv[2], "one")) {
+                sdsfree(server.masterhost);
```

Comment:
```suggestion
            sdsfree(server.masterhost);
            if (!strcasecmp(argv[1], "no") && !strcasecmp(argv[2], "one")) {
```
actually that leak existed before our fix